### ISSUE TITLE
Cannot send emails, postfix DNS issue

### ISSUE THREAD

**Issue by goldsziggy** [mailcow/mailcow-dockerized#817]
So, just starting to host my own mail server here, and I am running into an issue attempting to send emails.  I have narrowed it down to a DNS issue, I believe communicating between the container and unbound.

So first the error:
```
227C0460BAD: to=<OMITTED@gmail.com>, relay=none, delay=7991, delays=7991/0.36/0.28/0, dsn=4.4.3, status=deferred (Host or domain name not found. Name service error for name=gmail.com type=MX: Host not found, try again)
```
When attempting to send an email, I am see the above in the error logs.  Attempting to isolate the issue, I was going to install `ping` or `dig` onto the container when I noticed the following error as well:
```
root@host:/# apt-get update
Err:1 http://deb.debian.org/debian stretch InRelease
  Temporary failure resolving 'deb.debian.org'
Err:2 http://security.debian.org stretch/updates InRelease
  Temporary failure resolving 'security.debian.org'
Err:3 http://deb.debian.org/debian stretch-updates InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists... Done
W: Failed to fetch http://deb.debian.org/debian/dists/stretch/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://deb.debian.org/debian/dists/stretch-updates/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://security.debian.org/dists/stretch/updates/InRelease  Temporary failure resolving 'security.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
```
As simple GET requests are not able to resolve either, I started thinking it was a DNS issue.  To ensure it is not isolated to the container itself, I tried one of my personal containers. My personal container works perfectly fine.  When I attempted to do the same on another container in the mailcow family I saw the same issue.

Going further, I started looking at the docker-compose file, and noticed all containers are using the unbound container as the DNS.  Attaching my shell into that container I had no problem pinging both: `8.8.8.8` and `google.com`.  So the container itself does not seem to have any issues, but the requests routing do seem to have an issue.

I am stuck here, and am quite unsure on how to proceed.  Any help would be greatly appreciated!

**Comment by mkuron** [comment]
Probably your host firewall or your ISP are blocking requests to UDP port 53 (maybe with the exception of some well-known servers like 8.8.8.8). Check `iptables --list`, your ISP‘s documentation and `docker-compose logs unbound-mailcow`.

**Comment by goldsziggy** [comment]
So I did contact my ISP yesterday, and they informed me that they are not blocking any requests.  

Here are the results for the following commands

`iptables -list`
```
Chain INPUT (policy DROP)
target     prot opt source               destination
ufw-before-logging-input  all  --  anywhere             anywhere
ufw-before-input  all  --  anywhere             anywhere
ufw-after-input  all  --  anywhere             anywhere
ufw-after-logging-input  all  --  anywhere             anywhere
ufw-reject-input  all  --  anywhere             anywhere
ufw-track-input  all  --  anywhere             anywhere

Chain FORWARD (policy DROP)
target     prot opt source               destination
DOCKER-USER  all  --  anywhere             anywhere
DOCKER-ISOLATION  all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ufw-before-logging-forward  all  --  anywhere             anywhere
ufw-before-forward  all  --  anywhere             anywhere
ufw-after-forward  all  --  anywhere             anywhere
ufw-after-logging-forward  all  --  anywhere             anywhere
ufw-reject-forward  all  --  anywhere             anywhere
ufw-track-forward  all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
ufw-before-logging-output  all  --  anywhere             anywhere
ufw-before-output  all  --  anywhere             anywhere
ufw-after-output  all  --  anywhere             anywhere
ufw-after-logging-output  all  --  anywhere             anywhere
ufw-reject-output  all  --  anywhere             anywhere
ufw-track-output  all  --  anywhere             anywhere

Chain DOCKER (6 references)
target     prot opt source               destination
ACCEPT     tcp  --  anywhere             172.17.0.2           tcp dpt:2368
ACCEPT     tcp  --  anywhere             172.17.0.3           tcp dpt:6379
ACCEPT     tcp  --  anywhere             172.17.0.4           tcp dpt:9000
ACCEPT     tcp  --  anywhere             172.17.0.6           tcp dpt:9980
ACCEPT     tcp  --  anywhere             172.21.0.7           tcp dpt:https
ACCEPT     tcp  --  anywhere             172.21.0.7           tcp dpt:http
ACCEPT     tcp  --  anywhere             172.17.0.7           tcp dpt:59125
ACCEPT     tcp  --  anywhere             172.22.1.5           tcp dpt:submission
ACCEPT     tcp  --  anywhere             172.22.1.5           tcp dpt:urd
ACCEPT     tcp  --  anywhere             172.22.1.5           tcp dpt:smtp
ACCEPT     tcp  --  anywhere             172.22.1.7           tcp dpt:12345
ACCEPT     tcp  --  anywhere             172.22.1.7           tcp dpt:sieve
ACCEPT     tcp  --  anywhere             172.22.1.7           tcp dpt:pop3s
ACCEPT     tcp  --  anywhere             172.22.1.7           tcp dpt:imaps
ACCEPT     tcp  --  anywhere             172.22.1.7           tcp dpt:imap2
ACCEPT     tcp  --  anywhere             172.22.1.7           tcp dpt:pop3
ACCEPT     tcp  --  anywhere             172.22.1.251         tcp dpt:20443
ACCEPT     tcp  --  anywhere             172.22.1.251         tcp dpt:20080

Chain DOCKER-ISOLATION (1 references)
target     prot opt source               destination
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
RETURN     all  --  anywhere             anywhere

Chain DOCKER-USER (1 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere

Chain ufw-after-forward (1 references)
target     prot opt source               destination

Chain ufw-after-input (1 references)
target     prot opt source               destination
ufw-skip-to-policy-input  udp  --  anywhere             anywhere             udp dpt:netbios-ns
ufw-skip-to-policy-input  udp  --  anywhere             anywhere             udp dpt:netbios-dgm
ufw-skip-to-policy-input  tcp  --  anywhere             anywhere             tcp dpt:netbios-ssn
ufw-skip-to-policy-input  tcp  --  anywhere             anywhere             tcp dpt:microsoft-ds
ufw-skip-to-policy-input  udp  --  anywhere             anywhere             udp dpt:bootps
ufw-skip-to-policy-input  udp  --  anywhere             anywhere             udp dpt:bootpc
ufw-skip-to-policy-input  all  --  anywhere             anywhere             ADDRTYPE match dst-type BROADCAST

Chain ufw-after-logging-forward (1 references)
target     prot opt source               destination
LOG        all  --  anywhere             anywhere             limit: avg 3/min burst 10 LOG level warning prefix "[UFW BLOCK] "

Chain ufw-after-logging-input (1 references)
target     prot opt source               destination
LOG        all  --  anywhere             anywhere             limit: avg 3/min burst 10 LOG level warning prefix "[UFW BLOCK] "

Chain ufw-after-logging-output (1 references)
target     prot opt source               destination

Chain ufw-after-output (1 references)
target     prot opt source               destination

Chain ufw-before-forward (1 references)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
ACCEPT     icmp --  anywhere             anywhere             icmp destination-unreachable
ACCEPT     icmp --  anywhere             anywhere             icmp source-quench
ACCEPT     icmp --  anywhere             anywhere             icmp time-exceeded
ACCEPT     icmp --  anywhere             anywhere             icmp parameter-problem
ACCEPT     icmp --  anywhere             anywhere             icmp echo-request
ufw-user-forward  all  --  anywhere             anywhere

Chain ufw-before-input (1 references)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
ufw-logging-deny  all  --  anywhere             anywhere             ctstate INVALID
DROP       all  --  anywhere             anywhere             ctstate INVALID
ACCEPT     icmp --  anywhere             anywhere             icmp destination-unreachable
ACCEPT     icmp --  anywhere             anywhere             icmp source-quench
ACCEPT     icmp --  anywhere             anywhere             icmp time-exceeded
ACCEPT     icmp --  anywhere             anywhere             icmp parameter-problem
ACCEPT     icmp --  anywhere             anywhere             icmp echo-request
ACCEPT     udp  --  anywhere             anywhere             udp spt:bootps dpt:bootpc
ufw-not-local  all  --  anywhere             anywhere
ACCEPT     udp  --  anywhere             224.0.0.251          udp dpt:mdns
ACCEPT     udp  --  anywhere             239.255.255.250      udp dpt:1900
ufw-user-input  all  --  anywhere             anywhere

Chain ufw-before-logging-forward (1 references)
target     prot opt source               destination

Chain ufw-before-logging-input (1 references)
target     prot opt source               destination

Chain ufw-before-logging-output (1 references)
target     prot opt source               destination

Chain ufw-before-output (1 references)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
ufw-user-output  all  --  anywhere             anywhere

Chain ufw-logging-allow (0 references)
target     prot opt source               destination
LOG        all  --  anywhere             anywhere             limit: avg 3/min burst 10 LOG level warning prefix "[UFW ALLOW] "

Chain ufw-logging-deny (2 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere             ctstate INVALID limit: avg 3/min burst 10
LOG        all  --  anywhere             anywhere             limit: avg 3/min burst 10 LOG level warning prefix "[UFW BLOCK] "

Chain ufw-not-local (1 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere             ADDRTYPE match dst-type LOCAL
RETURN     all  --  anywhere             anywhere             ADDRTYPE match dst-type MULTICAST
RETURN     all  --  anywhere             anywhere             ADDRTYPE match dst-type BROADCAST
ufw-logging-deny  all  --  anywhere             anywhere             limit: avg 3/min burst 10
DROP       all  --  anywhere             anywhere

Chain ufw-reject-forward (1 references)
target     prot opt source               destination

Chain ufw-reject-input (1 references)
target     prot opt source               destination

Chain ufw-reject-output (1 references)
target     prot opt source               destination

Chain ufw-skip-to-policy-forward (0 references)
target     prot opt source               destination
DROP       all  --  anywhere             anywhere

Chain ufw-skip-to-policy-input (7 references)
target     prot opt source               destination
DROP       all  --  anywhere             anywhere
```


`docker-compose logs unbound-mailcow`
```
Attaching to mailcowdockerized_unbound-mailcow_1
unbound-mailcow_1    | Receiving anchor key...
unbound-mailcow_1    | Receiving root hints...
######################################################################## 100.0%
unbound-mailcow_1    | Receiving anchor key...
unbound-mailcow_1    | Receiving root hints...
######################################################################## 100.0%
unbound-mailcow_1    | Receiving anchor key...
unbound-mailcow_1    | Receiving root hints...
######################################################################## 100.0%
unbound-mailcow_1    | Receiving anchor key...
unbound-mailcow_1    | Receiving root hints...
######################################################################## 100.0%
unbound-mailcow_1    | [1513039870] unbound[1:0] warning: unbound is already running as pid 1.
unbound-mailcow_1    | Receiving anchor key...
unbound-mailcow_1    | Receiving root hints...
######################################################################## 100.0%
unbound-mailcow_1    | Receiving anchor key...
unbound-mailcow_1    | Receiving root hints...
######################################################################## 100.0%
```

Please let me know if you see anything out of the ordinary,
thanks!

**Comment by mkuron** [comment]
I didn‘t find anything unusual in there, but just to be sure, you should disable the ufw firewall on your host and reboot. Docker and firewalls don’t always get along flawlessly.

Also, can you run `docker-compose logs -t unbound-mailcow` so that we can see whether the log times correspond to you restarting the service or the service crashing and restarting periodically?

**Comment by goldsziggy** [comment]
So here is the results of the commands docker-compose logs -t unbound-mailcow
```
Attaching to mailcowdockerized_unbound-mailcow_1
unbound-mailcow_1    | 2017-12-13T20:47:15.415746915Z Receiving anchor key...
unbound-mailcow_1    | 2017-12-13T20:47:23.270746179Z Receiving root hints...
######################################################################## 100.0%
```

I have since attempted this guide for setup.
https://steemit.com/programming/@clutteredcode/docker-behind-ufw-firewall

The above helped for a separate issue i had, receiving mail..  So now my mailserver can recieve mail, but I am still seeing the same error for above for sending.

I don't believe anything is blocking on the ISP level, for when I run `dig mx gmail.com` on the host machine, it has no problem with resolution.
```
; <<>> DiG 9.10.3-P4-Ubuntu <<>> mx gmail.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 248
;; flags: qr rd ra; QUERY: 1, ANSWER: 5, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;gmail.com.			IN	MX

;; ANSWER SECTION:
gmail.com.		3440	IN	MX	10 alt1.gmail-smtp-in.l.google.com.
gmail.com.		3440	IN	MX	40 alt4.gmail-smtp-in.l.google.com.
gmail.com.		3440	IN	MX	30 alt3.gmail-smtp-in.l.google.com.
gmail.com.		3440	IN	MX	20 alt2.gmail-smtp-in.l.google.com.
gmail.com.		3440	IN	MX	5 gmail-smtp-in.l.google.com.

;; Query time: 26 msec
;; SERVER: 192.168.55.1#53(192.168.55.1)
;; WHEN: Fri Dec 15 05:39:18 CST 2017
;; MSG SIZE  rcvd: 161
```

**Comment by maffo999** [comment]
Can you try disabling the firewall completely for one time? That way you can at least find out if the issue lies there or not.
Also, what machine has IP 192.168.55.1? Is that a local DNS you run? IF so, have you tried using Google DNS or OpenDNS for a try?

**Comment by igloocube** [comment]
Check out my issue about DNS issues, may give you some pointers: https://github.com/mailcow/mailcow-dockerized/issues/850

TLDR: UDP was blocked. Dig will switch to using TCP so it works as you display where unbound does not.

**Comment by aureq** [comment]
If this is still happening, look at https://github.com/mailcow/mailcow-dockerized/issues/850#issuecomment-561572912
